src/sentry/search/events/fields.py

```diff
@@ -378,17 +378,24 @@ def normalize_count_if_value(args: Mapping[str, str]) -> Union[float, str]:
     value = args["value"]
     if column == "transaction.duration" or is_measurement(column) or is_span_op_breakdown(column):
         try:
-            value = float(value.strip("'"))
+            normalized_value = float(value.strip("'"))
         except Exception:
             raise InvalidSearchQuery(f"{value} is not a valid value to compare with {column}")
-
+    elif column == "transaction.status":
+        code = SPAN_STATUS_NAME_TO_CODE.get(value.strip("'"))
+        if code is None:
+            raise InvalidSearchQuery(f"{value} is not a valid value for transaction.status")
+        normalized_value = int(code)
```

Comment:
is code always expected to be an integer? Do we need to wrap this in a try/except to tell us if it's not a valid type?

Reply:
Added a check to be safe, but this shouldn't need the try/except since `SPAN_STATUS_NAME_TO_CODE` should give us something that's safe to int cast